### ISSUE TITLE
VectorBuilder#addAll(Vector) should perform a "shallow copy" of the argument, when the builder is currently empty.

### ISSUE THREAD

**Issue by joshlemer** [scala/bug#11329]
Take this example
```scala
val v = Vector.fill(1000000)(1)

val b = Vector.newBuilder

b.addAll(v)
```

Here, `b` will perform a deep copy of `v`, but it should be able to just do something along the lines of `this.initFrom(v)` and whatever other bookkeeping is required within the builder. 

This will be a really great improvement to performance, especially because this is how Vectors currently perform concatenation in most cases (`newBuilder.addAll(this).addAll(that).result()`)

**Comment by julienrf** [comment]
Should we really optimize for all these corner cases?

**Comment by joshlemer** [comment]
@julienrf I think that it's pretty valuable in this case, because of the fact that this is the way that Vector's currently concatenate with each other.

**Comment by dwijnand** [comment]
Note this shipped in 2.13.0-RC1 and was reverted for 2.13.1.

I _presume_ this improvement comes if/when https://github.com/scala/scala/pull/8534 is merged.
